scheds/rust/scx_rusty/src/bpf/main.bpf.c

```diff
@@ -1592,24 +1612,23 @@ s32 BPF_STRUCT_OPS(rusty_init_task, struct task_struct *p,
 {
 	u64 now = bpf_ktime_get_ns();
 	struct task_ctx taskc = {
-		.dom_active_pids_gen = -1,
+		.dom_active_task_ids_gen = -1,
 		.last_blocked_at = now,
 		.last_woke_at = now,
 		.preferred_dom_mask = 0,
 
 	};
 	struct task_ctx *map_value;
 	long ret;
-	pid_t pid;
-
-	pid = p->pid;
+	u64 task_id;
+	task_id = (u64)p;
 
 	/*
 	 * XXX - We want BPF_NOEXIST but bpf_map_delete_elem() in .disable() may
 	 * fail spuriously due to BPF recursion protection triggering
 	 * unnecessarily.
 	 */
-	ret = bpf_map_update_elem(&task_data, &pid, &taskc, 0 /*BPF_NOEXIST*/);
+	ret = bpf_map_update_elem(&task_data, &task_id, &taskc, 0 /*BPF_NOEXIST*/);
```

Comment:
Was BPF okay with this? It sometimes complains about converting pointers to scalar.